src/util/TestCaseExcelExportUtil.ts

```diff
@@ -9,18 +9,113 @@ import {
   Measure,
   TestCase,
   Group,
+  GroupPopulation,
   TestCaseExcelExportDto,
   PopulationDto,
   DefinitionDto,
   FunctionDto,
+  StratificationDto,
   TestCaseExecutionResultDto,
+  GroupedStratificationDto,
 } from "@madie/madie-models";
 import { CqlDefinitionCallstack } from "../components/editTestCase/groupCoverage/QiCoreGroupCoverage";
 import {
   StatementCoverageResult,
   buildHighlightingForGroups,
 } from "./cqlCoverageBuilder/CqlCoverageBuilder";
 
+const getTestCasesByGroup = (measure: Measure, groupId: string): TestCase[] => {
+  const testCasesByGroup: TestCase[] = [];
+  measure.testCases?.forEach((testCase) => {
+    testCase.groupPopulations?.forEach((groupPopulation) => {
+      if (groupPopulation.groupId === groupId) {
+        testCasesByGroup.push(testCase);
```

Comment:
ok I am little confused here, why do we have to do this? Doesn't every test case will have that groupPopulation ?

Reply:
@RohitKandimalla We need to group test cases by population criteria for excel export

Reply:
ok So we are returning a testCase object for every group ? If so why does testCasesByGroup have to be a List of TestCase, there will only be 1 group in a measure that matches the groupPopulations in its corresponding test case.

Reply:
`getTestCasesByGroup(measure, group.id);`
So this method is trying to create a list of TestCases, the returned list of testCases are the ones which has that particular groupID as a groupPopulation, but My question is, how is this retuned list will be different from measure.testcases ?

Essentially every testCase will have a groupPopulation whose id will be matched with measure.groups

Reply:
@RohitKandimalla test case populations and stratifications will be grouped by groups/population criteria, so the dto will be: group1: [testcase1, testcase2], group2: [testcase1, testcase2]...